EventListener/InvalidationSubscriber.php

```diff
@@ -220,6 +226,26 @@ private function invalidateRoutes(array $routes, Request $request)
                 }
             }
 
+            if (null === $route->getParams()) {
+                $route = $this->router->getRouteCollection()->get($route->getName());
```

Comment:
your code snippets don't do anything to prove @ddeboer is wrong (and this is logical, because he is not)